hedera-node/src/main/java/com/hedera/services/state/logic/BlockManager.java

```diff
@@ -84,12 +88,65 @@ public long getCurrentBlockNumber() {
 		return networkCtx.get().getManagedBlockNo();
 	}
 
+	public long getProvisionalBlockNumber() {
+		return provisionalBlockNumber;
+	}
+
+	public org.hyperledger.besu.datatypes.Hash getProvisionalBlockHash(long blockNumber) {
+		if (blockNumber == provisionalBlockNumber) {
+			return provisionalBlockHash;
+		}
+		return networkCtx.get().getBlockHashByNumber(blockNumber);
+	}
+
+	/**
+	 *
+	 * @param timestamp
+	 * @param gasLimit
+	 * @return
+	 */
+	public HederaBlockValues createProvisionalBlockValues(@NotNull final Instant timestamp, long gasLimit) {
+		var curNetworkCtx = networkCtx.get();
+
+		var blockNo = curNetworkCtx.getBlockNo();
+
+		// Before the 0.26 upgrade, we used the consensus timestamp as the block number; and if we
+		// get a zero block number, it means the post-0.26 block sync hasn't happened yet
+		Instant provisionalBlockTimestamp;
+		if (blockNo == 0) {
+			provisionalBlockNumber = timestamp.getEpochSecond();
+			provisionalBlockTimestamp = Instant.EPOCH;
```

Comment:
⚠️ There are two gotchas here:

1. There's a fix in `misc-3187-fixes` to get `blockNo` to be correctly returned as 0 **until a renumbering has happened**---so we might start by merging that branch (should be few or no conflicts).
2. We need to use `timestamp.getEpochSecond()` as the `provisionalBlockTimestamp`, not `Instant.EPOCH` here---otherwise an exciting variety of precompiles will blow up. 😁